### ISSUE TITLE
Make Aquaterm.app automatic search more robust

### ISSUE THREAD

**Issue by abgandar** [AquaTerm/AquaTerm#9]
Currently, the AquaTerm framework tries to find AquaTerm.app in this order (see - (BOOL)launchServer in AQTClientManager.m):
1) using the environment variable AQUATERM_PATH
2) using hard coded "/Applications/AquaTerm.app"
3) using Apple Launch Services based on the app creator code (AqTS)

For various uses it may be advantageous to  make the default path (option 2 above) a configure option so it can be overwritten at compile time. Then e.g. MacPorts could use /Applications/MacPorts/AquaTerm.app instead. That way, applications linked to the MacPorts libraries in /opt/local/Framework/AquaTerm.framework (or whatever it is) will call /Applications/MacPorts/AquaTerm.app, while apps linked to the installation from the default package in /Library/Frameworks/Aquaterm.framework will try to use /Applications/AquaTerm.app.

Related to this, the auto detection could be made more robust by also checking the version of the AquaTerm.app that is found to make sure it is the same as the library as different versions are not compatible and can lead to deadlock.

**Comment by mojca** [comment]
I fully agree with this. I even consider it a high priority to fix this. From an earlier discussion:

> > Even worse is that if I delete /Applications/AquaTerm.app it takes
> > AquaTerm.app from some random location when I run Gnuplot,
> > for example from
> > ~/dev/gnuplot/aquaterm/build/Deployment/AquaTerm.app
> > 
> > I don't like that weird random behaviour at all.
> 
> The way it is implemented is to see if the environment variable AQUATERM_PATH is set,
> if it is then use that to find the app, otherwise have the system search for the first instance it can find.
> IIRC the order is:
> ~/Applications
> /Applications
> "rest of system"
> 
> MacPorts might modiify that order though.

**Comment by mojca** [comment]
While fixing this we might want to consider something else as well: what should happen if system AquaTerm is already opened and then gnuplot from MacPorts tries to plot? Should another instance be opened or not?

Also: do we want to fix this before or after release of 1.1.1? Related issue: https://github.com/AquaTerm/AquaTerm/issues/8

**Comment by bpabbott** [comment]
I think Octave opens an instance of gnuplot for each plot window.

This isn't a problem as gnuplot is a command-line application. If an instance of Aquaterm is opened for each gnuplot, then the dock can become populated with a lot of Aquaterm icons.
